pandas/core/generic.py

```diff
@@ -2412,6 +2412,8 @@ def to_hdf(
         complib: Optional[str] = None,
         append: bool_t = False,
         format: Optional[str] = None,
+        min_itemsize=None,
```

Comment:
could these keywords added here and in #29939 for the public API be keyword only, xref #27544

Reply:
i like this idea, will do in a dedicated PR